packages/flutter/lib/src/widgets/restoration_properties.dart

```diff
@@ -0,0 +1,250 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+
+import 'editable_text.dart';
+import 'restoration.dart';
+
+/// A [RestorableProperty] that makes the wrapped value accessible to the owning
+/// [State] object via the [value] getter and setter.
+///
+/// Whenever a new [value] is set, [didUpdateValue] is called. Subclasses should
+/// call [notifyListeners] from this method if the new value changes what
+/// [toPrimitives] returns.
+///
+/// See also:
+///
+///  * [RestorableProperty], which is the super class of this class.
+///  * [RestorationMixin], to which a [RestorableValue] needs to be registered
+///    in order to work.
+///  * [RestorationManager], which provides an overview of how state restoration
+///    works in Flutter.
+abstract class RestorableValue<T> extends RestorableProperty<T> {
```

Comment:
Should this implement ValueNotifier<T>? As it fully matches the prototype

Reply:
The semantics are slightly different, though. `ValueNotifier` notifies when `value` changes. This one notifies when `toPrimitives` or `enabled` changes. And you can construct cases where `value` changes, but `toPrimitives` or `enabled` doesn't.